staging/src/k8s.io/component-base/config/testing/apigroup.go

```diff
@@ -0,0 +1,205 @@
+func VerifyExternalTypePackage(pkginfo *ComponentConfigPackage) error {
+	scheme, err := setup(pkginfo)
+	if err != nil {
+		return fmt.Errorf("test setup error: %v", err)
+	}
+	extraFns := map[string]testingFunc{
+		verifyExternalAPIVersion: verifyExternalAPIVersionFunc,
+	}
+	return runFuncs(scheme, pkginfo, extraFns)
+}
+
+// VerifyInternalTypePackage tests if internal component config package is defined correctly
+// Test tag naming (no tags allowed)
+// Test that GroupName has the k8s.io suffix
+// Test that GroupName == SchemeGroupVersion.GroupName
+// API version should be internal
+// Test that addKnownTypes and AddToScheme registers at least one type and doesn't error
+// Test that the GroupName is named correctly (based on ComponentName), and there is a {Component}Configuration kind in the scheme
+func VerifyInternalTypePackage(pkginfo *ComponentConfigPackage) error {
+	scheme, err := setup(pkginfo)
+	if err != nil {
+		return fmt.Errorf("test setup error: %v", err)
+	}
+	extraFns := map[string]testingFunc{
+		verifyInternalAPIVersion: verifyInternalAPIVersionFunc,
+	}
+	return runFuncs(scheme, pkginfo, extraFns)
+}
+
+func setup(pkginfo *ComponentConfigPackage) (*runtime.Scheme, error) {
+	if len(pkginfo.ComponentName) == 0 ||
+		len(pkginfo.GroupName) == 0 ||
+		pkginfo.SchemeGroupVersion.Empty() ||
+		pkginfo.AddToScheme == nil {
+		return nil, fmt.Errorf("invalid argument: not all parameters were passed correctly to the function")
+	}
+
+	scheme := runtime.NewScheme()
+	if err := pkginfo.AddToScheme(scheme); err != nil {
+		return nil, fmt.Errorf("AddToScheme must not return an error")
+	}
+	if len(scheme.AllKnownTypes()) == 0 {
+		return nil, fmt.Errorf("AddToScheme doesn't register any type")
+	}
+	return scheme, nil
+}
+
+func runFuncs(scheme *runtime.Scheme, pkginfo *ComponentConfigPackage, extraFns map[string]testingFunc) error {
+	verifyFns := []testingFunc{}
+	for name, fn := range testingFuncs {
+		if pkginfo.SkipTests.Has(name) {
+			continue
+		}
+		verifyFns = append(verifyFns, fn)
+	}
+	for name, fn := range extraFns {
+		if pkginfo.SkipTests.Has(name) {
+			continue
+		}
+		verifyFns = append(verifyFns, fn)
+	}
+	errs := []error{}
+	for _, fn := range verifyFns {
+		if err := fn(scheme, pkginfo); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return errors.NewAggregate(errs)
+}
+
+func verifyTagNamingFunc(scheme *runtime.Scheme, pkginfo *ComponentConfigPackage) error {
+	return apinamingtest.VerifyTagNaming(scheme, pkginfo.AllowedTags, pkginfo.AllowedNonstandardJSONNames)
+}
+
+func verifyGroupNameSuffixFunc(scheme *runtime.Scheme, _ *ComponentConfigPackage) error {
+	return apinamingtest.VerifyGroupNames(scheme, sets.NewString())
+}
+
+func verifyGroupNameMatchFunc(_ *runtime.Scheme, pkginfo *ComponentConfigPackage) error {
+	if pkginfo.GroupName != pkginfo.SchemeGroupVersion.Group {
+		return fmt.Errorf("GroupName must equal SchemeGroupVersion.Group")
```

Comment:
Also include the input in the error here (double-check the other functions too).